src/coreclr/jit/llvm.cpp

```diff
@@ -2607,10 +2689,31 @@ void Llvm::lowerCallToShadowStack(GenTreeCall* callNode)
     }
 }
 
+bool Llvm::independentCanBeReplacedWithItsField(LclVarDsc* addrVarDsc)
+{
+    return _compiler->lvaGetPromotionType(addrVarDsc) == Compiler::PROMOTION_TYPE_INDEPENDENT &&
+           addrVarDsc->CanBeReplacedWithItsField(_compiler);
+}
+
 void Llvm::lowerStoreLcl(GenTreeLclVarCommon* storeLclNode)
 {
     LclVarDsc* addrVarDsc = _compiler->lvaGetDesc(storeLclNode->GetLclNum());
 
+    if (independentCanBeReplacedWithItsField(addrVarDsc))
+    {
+        ClassLayout* layout      = addrVarDsc->GetLayout();
+        GenTree*     data        = storeLclNode->gtGetOp1();
+        var_types    addrVarType = addrVarDsc->TypeGet();
+
+        storeLclNode->SetOper(GT_LCL_VAR_ADDR);
+        storeLclNode->ChangeType(TYP_I_IMPL);
+        storeLclNode->SetLclNum(addrVarDsc->lvFieldLclStart + 0);
```

Comment:
```suggestion
        storeLclNode->SetLclNum(addrVarDsc->lvFieldLclStart);
```
Nit: to be consistent with the rest of compiler code.